packages/js/product-editor/src/blocks/generic/number/edit.tsx

```diff
@@ -47,31 +48,40 @@ export function Edit( {
 				value &&
 				parseFloat( value ) < min
 			) {
-				return sprintf(
-					// translators: %d is the minimum value of the number input.
-					__(
-						'Value must be greater than or equal to %d',
-						'woocommerce'
+				return {
+					message: sprintf(
+						// translators: %d is the minimum value of the number input.
+						__(
+							'Value must be greater than or equal to %d',
+							'woocommerce'
+						),
+						min
 					),
-					min
-				);
+					context: clientId,
```

Comment:
You could go one step further, and instead of passing in the `clientId` as part of the context you could remove this and make use of [`useBlockEditContext`](https://github.com/WordPress/gutenberg/blob/4f706f1777d05566f66b80a82560dd7c498942dc/packages/block-editor/README.md#useblockeditcontext) hook within the `useValidation` hook.
This `useBlockEditContext` provides you the `clientId`. So calling that within the `useValidation` should provide you the `clientId` of the relevant block.

Reply:
I'm resolving this comment because we decided to keep it as it is now.